### PR TITLE
[RootSignature] Add `obj2yaml` testing

### PR DESCRIPTION
**Pull request by inbelic**
This pr adds testing using `obj2yaml` for root signatures.

There are types of tests added:
- edge-cases: which check for edge cases as specified in the test cases
- consistency: which checks that `DXC` and `clang` set the root signature values to be the same. The particular values are not necessarily important, as they will have been covered in the edge-cases 

Resolves: https://github.com/llvm/offload-test-suite/issues/260.

### CODE REVIEW

**Review by llvm-beanz — APPROVED** [review]
This LGTM. I wonder if longer term we would be better off testing this with a more compact text representation supported via `llvm-objdump` (https://github.com/llvm/llvm-project/issues/152526).

File: test/Feature/RootSignatures/RootDescriptorAndTables.test [hunk 1]
```diff
@@ -64,6 +64,38 @@ DescriptorSets:
...
+# OBJ:   RootParametersOffset: 24
+# OBJ:   NumStaticSamplers: 0
+# OBJ:   StaticSamplersOffset: 116
+# OBJ:   Parameters:
+# OBJ: - ParameterType:   2
+# OBJ:   ShaderVisibility: 0
+# OBJ:   Descriptor:
+# OBJ:   RegisterSpace:   0
+# OBJ:   ShaderRegister:  2
+# OBJ: - ParameterType:   0
```
Comment (bogner) [inline test/Feature/RootSignatures/RootDescriptorAndTables.test:86]: Looks like we aren't checking everything (hence not using CHECK-NEXT I suppose), which is fine, but how did you decide which fields are interesting?
